src/app/shared/infinity-scroll/infinity-scroll.directive.ts

```diff
@@ -0,0 +1,42 @@
+import {Directive, EventEmitter, HostListener, Output} from '@angular/core';
+import {LoadDirection} from '../load-direction/load-direction.enum';
+
+@Directive({
+    selector: '[appInfinityScroll]',
+})
+export class InfinityScrollDirective {
+    @Output() readonly loadData = new EventEmitter<LoadDirection>();
+
+    private readonly borderOffset = 100;
+    private lastScrollTop = 0;
+
+    @HostListener('scroll', ['$event', '$event.target']) onScroll(
+        event: MouseEvent,
+        elem: HTMLElement,
+    ) {
+        this.borderObserver(elem);
+        event.stopPropagation();
+    }
```

Comment:
Интересный подход 🤔, но кажется, что можно было бы просто обратиться к свойству)
```suggestion
    @HostListener('scroll', ['$event']) onScroll(event: MouseEvent) {
        this.borderObserver(event.target);
        event.stopPropagation();
    }
```

А вообще, нужно ли останавливать всплытие события?

Reply:
а ну да, срколл не вызывает всплытие, но тогда и смысла в event нет. Тогда я сразу получу ['$event.target']